pkg/ingester/activeseries/matchers.go

```diff
@@ -35,13 +35,16 @@ func (m *Matchers) Config() CustomTrackersConfig {
 	return m.cfg
 }
 
-func (m *Matchers) Matches(series labels.Labels) []bool {
+// Matches returns a []int containing only matcher indexes which are matching
+func (m *Matchers) Matches(series labels.Labels) []int {
 	if len(m.matchers) == 0 {
 		return nil
 	}
-	matches := make([]bool, len(m.matchers))
+	matches := []int{}
 	for i, sm := range m.matchers {
-		matches[i] = sm.Matches(series)
+		if sm.Matches(series) {
+			matches = append(matches, i)
+		}
```

Comment:
Can we see benchmarks of this? :)

This will have to allocate a new slice every `2^n` matchers, which is probably more expensive than just allocating a slice of bools since it will push unnecessary pressure on the GC.

Reply:
If we're worried about memory consumption here, then IMO we should keep the previous approach and change `[]bool` to a bitarray backed by `[]uint64`